### PR TITLE
Alleviate bottleneck in removeAccepted

### LINKED ISSUE #997: Achieve 50 TPS or 300 txs per a block in test harness
Currently, test harness reports up to 10 TPS which in most cases means  around 56 transactions accepted within 5-6 block time.

Obviously, the main limitation comes from two constants we've got:

```go

// MaxTxSetSize defines the maximum amount of transactions.
// It is TBD along with block size and processing.MaxFrameSize.
const MaxTxSetSize = 150000

// MaxFrameSize is set at 250000 bytes.
MaxFrameSize = uint64(250000)

```

### LINKED ISSUE #1003: Alleviate the processing bottleneck in the mempool, when processing an accepted block
@goshawk-3 observed this when running with a `batchSize` of 1000:

```
time="2021-03-31T12:26:59+03:00" level=info msg="Measured TPS 3.60 (18/5)" cumulative_tx_count=18 height=7
time="2021-03-31T12:27:03+03:00" level=info msg="Measured TPS 6.20 (31/5)" cumulative_tx_count=49 height=8
time="2021-03-31T12:27:11+03:00" level=info msg="Measured TPS 22.00 (132/6)" cumulative_tx_count=181 height=9
time="2021-03-31T12:27:47+03:00" level=info msg="Measured TPS 12.00 (252/21)" cumulative_tx_count=433 height=10
time="2021-03-31T12:28:24+03:00" level=info msg="Measured TPS 7.33 (286/39)" cumulative_tx_count=719 height=11
time="2021-03-31T12:29:41+03:00" level=info msg="Measured TPS 5.56 (306/55)" cumulative_tx_count=1025 height=12
time="2021-03-31T12:32:01+03:00" level=info msg="Measured TPS 2.62 (306/117)" cumulative_tx_count=1331 height=13
```

This should be investigated and potentially fixed.

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #997, fixes #1003

### CODE REVIEW

File: pkg/core/mempool/hashmap.go [hunk 1]
```diff
@@ -141,6 +141,24 @@ func (m *HashMap) Get(txID []byte) transactions.ContractCall {
...
 
+// Delete a key in the hashmap.
+func (m *HashMap) Delete(txID []byte) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	var k txHash
+
+	copy(k[:], txID)
+	delete(m.data, k)
```
Comment (goshawk-3) [inline pkg/core/mempool/hashmap.go:152]: It could be good also to recalculate `m.txsSize`.

```
size := m.data[k]
m.txsSize -= size
```

Comment (julesdesmit) [reply pkg/core/mempool/hashmap.go:155]: Good catch! Updated it :)

File: pkg/core/mempool/hashmap.go [hunk 2]
```diff
@@ -141,6 +141,27 @@ func (m *HashMap) Get(txID []byte) transactions.ContractCall {
...
+// Delete a key in the hashmap.
+func (m *HashMap) Delete(txID []byte) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	var k txHash
+
+	copy(k[:], txID)
+
+	m.txsSize -= uint32(m.data[k].size)
```
Comment (goshawk-3) [inline pkg/core/mempool/hashmap.go:153]: Would you also check if `m.data` contains `k`?

Comment (goshawk-3) [reply pkg/core/mempool/hashmap.go:153]: Sorry. Forgot to add this to the snippet.

Comment (julesdesmit) [reply pkg/core/mempool/hashmap.go:153]: Of course, good catch, missed that myself :)

Comment (julesdesmit) [reply pkg/core/mempool/hashmap.go]: Updated!

**Review by goshawk-3 — APPROVED** [review]

**Pull request merged**